test/test_train_mp_imagenet.py

```diff
@@ -199,6 +200,9 @@ def train_imagenet():
 
   device = xm.xla_device()
   model = get_model_property('model_fn')().to(device)
+  if pjrt.using_pjrt():
```

Comment:
What's this change about? To mimic DDP behavior?

Reply:
Added a comment here to explain. Because threads share the same random generator while initializing, it's impossible to set the same initialization seed across replicas in PJRT with TPU v2/v3, so you have to synchronize the parameters manually. DDP does not broadcast parameters according to the [documentation](https://pytorch.org/docs/stable/generated/torch.nn.parallel.DistributedDataParallel.html#torch.nn.parallel.DistributedDataParallel).

I included this change in this PR so I can use it as an example in `pjrt.md`.

Reply:
The issue here is that with TPU v2/v3, each process manages multiple replicas, which all share the same global RNG. For example, rank 0 will have ordinals 0 and 1 concurrently initializing and advancing the random generator, and you'll end up with different parameters between the two threads. On TPU v4, you can set the same seed in each process since each process runs initialization only once (assuming you are running with one TPU chip per process, which is not always true).

We could work around the limitation for TPU v2/v3 if we could 1) initialize the parameters directly on the device (we could set both TPU cores' seeds to the same thing) or 2) directly supply a separate RNG with the same seed for each thread. As far as I know, neither of these are possible.

My understanding from the DDP documentation is that parameters are not broadcast between replicas, but you can optionally broadcast buffers on each forward pass:

```
Parameters are never broadcast between processes. The module performs an all-reduce step on gradients and assumes that they will be modified by the optimizer in all processes in the same way. Buffers (e.g. BatchNorm stats) are broadcast from the module in process of rank 0, to all other replicas in the system in every iteration.
```

If DDP does broadcast parameters at intitialization, then I can move this line into the experimental implementation in `pjrt.py` to maintain parity.